Added wiki page on Beautiful Soup

chenbriann opened the pull request:
This guide is an introduction to using the Beautiful Soup parsing library for web scraping and extracting data from HTML in python. It provides a brief introduction to the library, installation instructions and how to use it to extract data.

jfitzgerald1126 reviewed (changes requested):
This guide is a good introduction for beginners to web scraping and using BS4 in Python. However, I think you could mention when a user would typically choose BS4 over a more complex and capable scraping tool like selenium, or vice versa. 

Furthermore, it would help for readability purposes to make your python code snippets more consistent. For example, you could include this code in the subsequent code blocks to show that it is continuous.
```python
import requests

# Get the HTML from the website
url = 'https://example.com'
response = requests.get(url)
```

chenbriann commented:
> This guide is a good introduction for beginners to web scraping and using BS4 in Python. However, I think you could mention when a user would typically choose BS4 over a more complex and capable scraping tool like selenium, or vice versa.
> 
> Furthermore, it would help for readability purposes to make your python code snippets more consistent. For example, you could include this code in the subsequent code blocks to show that it is continuous.
> 
> ```python
> import requests
> 
> # Get the HTML from the website
> url = 'https://example.com'
> response = requests.get(url)
> ```

Thanks for your suggestions! I added a paragraph explaining when to use bs4 over selenium as well as a link to your selenium guide. I also took into consideration your feedback on the code snippets and updated them accordingly. Furthermore, I forgot to add a link to my guide on the Tech Stacks page so I have also fixed that issue. Let me know if there are any other changes needed or if it's good to go!
